pkg/systemd/generate/systemdgen.go

```diff
@@ -94,6 +98,7 @@ Before={{- range $index, $value := .RequiredServices -}}{{if $index}} {{end}}{{
 {{- end}}
 
 [Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
```

Comment:
Can we template this to make sure the environment variable always matches EnvVariable above?